main/src/ca/uwaterloo/flix/language/ast/KindedAst.scala

```diff
@@ -89,6 +89,8 @@ object KindedAst {
 
     case class Apply(exp: Expr, exps: List[Expr], tvar: Type.Var, evar: Type.Var, loc: SourceLocation) extends Expr
 
+    case class ApplyDef(symUse: Ast.DefSymUse, exps: List[Expr], ftvar: Type, tvar: Type.Var, evar: Type.Var, loc: SourceLocation) extends Expr
```

Comment:
search for ftvar

Reply:
oh thought you only wanted ftpe because it's not instantiated at this point